src/test/test_protocol_nmea.cpp

```diff
@@ -0,0 +1,1283 @@
+    // cout << "After (" << abuf_n << "):\n" << abuf;
+
+    ASSERT_TRUE( abuf_n == buf.size() );
+    ASSERT_TRUE( memcmp(abuf, buf.c_str(), abuf_n) == 0 );
+}
+
+
+TEST(protocol_nmea_4p11, GSV_binary_GSV)
+{
+    nema_set_protocol_version(NMEA_PROTOCOL_4P10);
+
+    string buf;
+
+    // GPS & SBAS        #msgs,msg#,numSV,  svid,elv,azm,cno, ..., signalId*checksum
+    buf += "$GPGSV,4,1,14" ",02,40,310,43" ",08,07,324,31" ",10,48,267,45" ",15,37,053,45"  ",1" "*67\r\n";
+    buf += "$GPGSV,4,2,14" ",16,12,268,35" ",18,69,078,41" ",23,74,336,40" ",24,15,111,37"  ",1" "*62\r\n";
+    buf += "$GPGSV,4,3,14" ",26,02,239,31" ",27,35,307,38" ",29,12,162,37" ",32,14,199,39"  ",1" "*60\r\n";
+    buf += "$GPGSV,4,4,14" ",44,43,188,43" ",46,40,206,43"                                  ",1" "*65\r\n";
+    buf += "$GPGSV,3,1,09" ",10,48,267,45" ",15,37,053,26" ",18,69,078,34" ",23,74,336,34"  ",6" "*68\r\n";
+    buf += "$GPGSV,3,2,09" ",24,15,111,25" ",26,02,239,18" ",27,35,307,27" ",29,12,162,21"  ",6" "*64\r\n";
+    buf += "$GPGSV,3,3,09" ",32,14,199,25"                                                  ",6" "*58\r\n";
+    // Galileo
+    buf += "$GAGSV,1,1,04" ",05,65,144,41" ",09,39,052,43" ",34,71,341,42" ",36,46,105,39"  ",7" "*7E\r\n";
+    buf += "$GAGSV,1,1,04" ",05,65,144,30" ",09,39,052,30" ",34,71,341,27" ",36,46,105,30"  ",2" "*73\r\n";
+    // Beidou
+    buf += "$GBGSV,2,1,08" ",11,09,141,34" ",14,52,047,44" ",27,32,313,43" ",28,80,263,44"  ",1" "*71\r\n";
+    buf += "$GBGSV,2,2,08" ",33,81,039,43" ",41,43,230,42" ",43,33,148,42" ",58,,,44"       ",1" "*4E\r\n";
+    buf += "$GBGSV,1,1,02" ",11,09,141,16" ",14,52,047,32"                                  ",B" "*0D\r\n";
+    // QZSS
+    buf += "$GQGSV,1,1,01" ",02,45,101,30"                                                  ",1" "*54\r\n";
+    // GLONASS
+    buf += "$GLGSV,2,1,06" ",65,85,260,33" ",66,28,217,30" ",72,36,034,35" ",81,20,324,33"  ",1" "*75\r\n";
+    buf += "$GLGSV,2,2,06" ",87,47,127,35" ",88,73,350,34"                                  ",1" "*75\r\n";
+    buf += "$GLGSV,1,1,01" ",87,47,127,20"                                                  ",3" "*41\r\n";
+
+    uint32_t cnoSum = 0, cnoCount = 0;
+    gps_sat_t gpsSat = {};
+    gps_sig_t gpsSig = {};
+    char abuf[ASCII_BUF2] = { 0 };
+
+    for (char *ptr = (char*)(buf.c_str()); ptr < (buf.c_str() + buf.size()); )
+    {
+        ptr = nmea_parse_gsv(ptr, buf.size(), &gpsSat, &gpsSig, &cnoSum, &cnoCount);
+    }
+    int abuf_n = nmea_gsv(abuf, ASCII_BUF2, gpsSat, gpsSig);
+
+    // cout << "Before (" << buf.size() << "):\n" << buf;
+    // cout << "After (" << abuf_n << "):\n" << abuf;
+
+    ASSERT_TRUE( abuf_n == buf.size() );
+    ASSERT_TRUE( memcmp(abuf, buf.c_str(), abuf_n) == 0 );
+}
+
+
+#if 1
+TEST(protocol_nmea, binary_GSV_binary)
+{
+    gps_sat_t gpsSat = {};
+    gpsSat.numSats = 33;
+    gpsSat.timeOfWeekMs = 436693200;
+    gps_sat_sv_t *sat = &(gpsSat.sat[0]);
+    sat->azim = 310;
```

Comment:
These are a lot of magic numbers; I suspect these are coming from some observations to test against.  Is there a better way to setup this state, so its more apparent what is happening (or what its aiming to test)?